src/test/java/com/google/api/codegen/testdata/library_config.baseline

```diff
@@ -16,10 +16,12 @@ language_settings:
 interfaces:
 - name: google.example.library.v1.LibraryService
   collections:
+  - name_pattern: archives/{archive_path=**}/books/{book}
+    entity_name: book
```

Comment:
It is just the order they get processed in. I could change it, but I don't know how to guarantee that I'm not causing the same problem in some other proto layout. Could possibly collect all conflicting names and order them by simplicity somehow, but this feels like overkill. This can be changed manually by users in the generated config file, which they would probably want to do anyway (e.g. as in library_gapic.yaml)